docs/shared/reference/project-configuration.md

```diff
@@ -616,6 +616,20 @@ Additionally, when using the expanded object syntax, you can specify individual
 This configuration is usually not needed. Nx comes with reasonable defaults (imported in `nx.json`) which implement the
 configuration above.
 
+### Sync Generators
+
+In the same way that `dependsOn` tells Nx to run a task before this task, the `syncGenerator` property tells Nx to run a specific type of generator (aimed at keeping applicable configuration files in sync with your project graph) before the task is run. [Sync generators](/concepts/sync-generators) are available in Nx 19.7+.
```

Comment:
Same here about the sync generators not being limited to:

- update configuration files
- use the project graph as the source of truth